libs/librepcb/common/graphics/graphicsview.cpp

```diff
@@ -203,6 +203,39 @@ void GraphicsView::zoomAnimationValueChanged(const QVariant& value) noexcept {
  *  Inherited from QGraphicsView
  ******************************************************************************/
 
+bool GraphicsView::event(QEvent* event) {
+    if (event->type() == QEvent::NativeGesture) {
+        QNativeGestureEvent* gesture = dynamic_cast<QNativeGestureEvent*>(event);
+        switch (gesture->gestureType()) {
+          case Qt::BeginNativeGesture:
+          case Qt::EndNativeGesture: {
+            return true;
+          }
+          case Qt::ZoomNativeGesture: {
+            QTransform transform;
+            transform.scale(1+gesture->value(), 1+gesture->value());
+            setTransform(transform, true);
+            return true;
+          }
+          default: {
+            // Only process zoom gestures
+            return false;
+          }
+        }
+    }
+
+    return false;
+}
+
+// This function receives the wheel events for the viewport widget.
+void GraphicsView::wheelEvent(QWheelEvent* event) {
```

Comment:
Same here: Couldn't `QWheelEvent` be handled in `eventFilter()` instead?

Reply:
I've removed the main `event()` function, but I need to keep `wheelEvent()`. The reason is that when the `QWheelEvent` is translated into a `QGraphicsSceneWheelEvent`, it loses the source information, and so I cannot test for whether it's a synthetic event and reject it. Instead, I need to do it here and send it up to the `QAbstractScrollArea` if it's a trackpad event. Otherwise, it'll pass to the `eventFilter()`.